src/main/java/org/opensearch/knn/jni/JNIService.java

```diff
@@ -50,7 +52,12 @@ public static void createIndex(
         }
 
         if (KNNEngine.FAISS == knnEngine) {
-            FaissService.createIndex(ids, vectorsAddress, dim, indexPath, parameters);
+            if (parameters.get(KNNConstants.INDEX_DESCRIPTION_PARAMETER) != null
```

Comment:
This logic of identifying the binary index lets abstract this out in a utility. I think this is going to be used at other places too.

Reply:
As this is java side code, will make the change in the next PR.